src/psyclone/domain/nemo/transformations/nemo_arrayrange2loop_trans.py

```diff
@@ -351,8 +351,8 @@ def validate(self, node, options=None):
                     raise TransformationError(
                         "The config file specifies '{0}' as the name of the "
                         "iteration variable but this is already declared in "
-                        "the code as something that is not a scalar integer."
-                        "".format(loop_variable_name))
+                        "the code as something that is not a scalar integer, "
+                        "or is a deferred type.".format(loop_variable_name))
```

Comment:
Sorry to be a pain but please could you add a test for this specific case - I suspect that you will need to check on the `type` of `loop_variable_symbol` since only a `DataSymbol` has the `is_scalar` method.

Reply:
No, it's a fair comment. Actually DeferredType returns False if you call it with is_scalar, but that is not very nice behaviour to rely on so I've updated the if test as well as suggested.